storage/mongo/mongo.go

```diff
@@ -65,6 +66,22 @@ func (r *Backend) Open(options url.Values) {
 	r.mongoSession.SetMode(mgo.Monotonic, true)
 }
 
+func (r Backend) GetLastDataItems(tenant string, id string, numOfItems int) ([]storage.DataItem, error) {
+	res := make([]storage.DataItem, numOfItems)
+	sessionCopy := r.mongoSession.Copy()
+	defer sessionCopy.Close()
+
+	c := sessionCopy.DB(tenant).C(id)
+	if count, _ := c.Count(); count < numOfItems {
+		return nil, fmt.Errorf("Not Enough items in database: Found %d, Queried: %d", count, numOfItems)
+	}
+	err := c.Find(nil).Sort("-timestamp").Limit(numOfItems).All(&res)
```

Comment:
I would rather the caller function to handle the error than `GetLastDataItems` to do that.
In other words, just return the error:
```Go
   return c.Find(nil).Sort("-timestamp").Limit(numOfItems).All(&res)
```